src/data/toc.ts

```diff
@@ -54,3 +57,21 @@ function flattenTOC(arr: any[]) {
   return result;
 }
 
+// @ts-ignore
+function flattenTOCDeep(array: any[]) {
```

Comment:
This existing `flattenTOC` just keeps the ids. Why not adjust that function to store the entire item (and add the depth attribute), and then re-created the old `flattened` list of ids based on the complete flattened list? Or, where is the old flattened list used, and can you adjust it to use the complete flattened toc?

Reply:
Done